functions/index.js

```diff
@@ -133,28 +154,88 @@ exports.pullDataFromSftp= functions.https.onRequest((request, response) => {
-         // Next step is to write this information to the firebase db.
-         for (var i = 0;i<results.data.length ;i++) {
-           console.log("Entry number", i, ":");
-           console.log("---------------");
-           for (var key in results.data[i]) {
-               if(results.data[i][key] != "") {
-                 console.log("key " + key + " has value " + results.data[i][key]);
-               }
-               else {
-                 console.log("key " + key + " has no value ");
-               }
-           }
-         console.log("**************************************\n");
-         }
+  //papaparse (https://www.papaparse.com)returns 'results' which has an array 'data'.
+  // Each entry in 'data' is an object, a set of key/values that match the header at the head of the csv file.
+    Papa.parse(fileContent, {
+      header: true,
+      skipEmptyLines: true,
+      complete: function(results) {
+        console.log("Found "+ results.data.length + " lines in file content\n");
+        //printing all the key values in the csv file to console ** for now **
+        // Next step is to write this information to the firebase db.
+        for (var i = 0;i<results.data.length ;i++) {
+          console.log("Entry number", i, ":");
+          console.log("---------------");
+          var user = new User();
+          user.dateCreated = Date.now();
+          for (var key in results.data[i]) {
+              if(results.data[i][key] != "") {
+                console.log("key " + key + " has value " + results.data[i][key]);
+                switch (key) {
+                  case 'System Name ID':
+                    user.uid = results.data[i][key];
+                    console.log(user.uid);
+                    break;
+                   case 'First Name':
+                     user.firstName = results.data[i][key];
+                     console.log(user.firstName);
+                     break;
+                   case 'Last Name':
+                     user.lastName = results.data[i][key];
+                     console.log(user.lastName);
+                     break;
+                   case 'Email Address':
+                      user.email = isValidEmail(results.data[i][key])
+                      ? results.data[i][key].trim().toLowerCase()
+                      : null;
+                     console.log(user.email);
+                    break;
+                }
+              }
+              else {
+                console.log("key " + key + " has no value ");
+              }
+          }
+          updateOrCreateUseronFireStore (user)
+        console.log("**************************************\n");
+        }
+     }
+  });
+}
+
+function updateOrCreateUseronFireStore (user){
+  let usersCollection = db.collection('users');
+  usersCollection.where('uid', '==', user.uid).get()
+    .then(snapshot => {
+      if (snapshot.empty) {
+        console.log('No matching documents. Creating a new document for this user');
```

Comment:
```suggestion
        console.log('No matching documents. Creating a new document for this user');
          // TODO: create a validate function on the user model and use it to ensure that we only
          // push valid user info to Firestore. I'm not sure I trust Outcome Tracker's sftp dumps enough 
          // to assume that they will always be giving us all the info we need
```
I like making a habit of leaving `// TODO`s in the code, and creating corresponding issues in the github project when I think of something tangental to the current work.  Validating the user isn't a vital part of this PR, so it's a good 'todo' for later.  You _could_ do it as part of this PR, but keeping PRs and issues small makes it more likely that each one will be completed quickly, and also allows other developers to contribute little pieces too.